src/libraries/Common/src/Interop/Windows/BCrypt/Interop.BCryptAlgPseudoHandle.cs

```diff
@@ -19,5 +19,7 @@ public enum BCryptAlgPseudoHandle : uint
             BCRYPT_SHA512_ALG_HANDLE = 0x00000061,
             BCRYPT_PBKDF2_ALG_HANDLE = 0x00000331,
         }
+
+        internal static bool PseudoHandlesSupported { get; } = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 0);
```

Comment:
@GrabYourPitchforks  Does this need to be a `static readonly` field for the JIT do to its "ooh, I can fold this away" magic, or will it peek through the property like this?

Reply:
I'm far from an expert in this area but my cursory check on this seemed show that the jit will follow through to the field. 

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlPkBiA9m+QGYfkC85AHZwA7qw4AKAJQBuRszKUALOQCy0+vObMAll3ISebAHQB9YBARSt2zUVsPKeAJwSARFx1wANgBM3sjbaAL42ofbMNopYKgByGgwRDnoGRsYAQpbWSbaJjrZYLu4ADghsxQFyOSFhxOHE0TjibMR5ChQIcBC+bILeAJ7kwBze5OaW/EKiTsYAShCCPWDSxrFwAB4wEjhS/AJ4VdqKw2yjmQj05ADmcDAy5MGTwmKFcwtLK2ub27t8+1XBIA

Reply:
Yeah, that does seem promising.  Unfortunately, changing both the field and the property to static and setting them to false doesn't cause the emitted method body to be a nop for either the field case or property case... in both cases it loads the memory address, does a branch, and produces the unhittable call instruction.

Maybe it's a tiered JIT thing, or maybe it requires conditions I don't understand.

Reply:
It will work fine once the code gets to tier 1.  I _think_ sharplab is showing the equivalent of tiering disabled code, in which case some call sites may not be able to assume the static cctor has executed.

You can see this with a benchmark.net test like:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

[MemoryDiagnoser]
[DisassemblyDiagnoser]
public class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    public static bool StaticFalseProp { get; } = false;
    public static readonly bool StaticFalseField = false;

    [Benchmark]
    public int UseProp()
    {
        if (StaticFalseProp)
        {
            return 42;
        }

        return 84;
    }

    [Benchmark]
    public int UseField()
    {
        if (StaticFalseField)
        {
            return 42;
        }

        return 84;
    }
}
```
which outputs:
## .NET 6.0.0 (6.0.21.15406), X64 RyuJIT
```assembly
; Program.UseProp()
       mov       eax,54
       ret
; Total bytes of code 6
```

## .NET 6.0.0 (6.0.21.15406), X64 RyuJIT
```assembly
; Program.UseField()
       mov       eax,54
       ret
; Total bytes of code 6
```